### ISSUE TITLE
Feature Request: Paket Fetch

### ISSUE THREAD

**Issue by pms1969** [fsprojects/Paket#1412]
I've been thinking I'd like to dispense of nuget altoghether in favour of paket, but not having the ability to just fetch a nuget package (without the need for a paket.dependencies file) is holding me back.

So I propose a new command: "fetch" that does just that.  Usage would be something like:
paket fetch mycomponent.2.4.6 output mydirectory source http://myinternal.nuget.server:89 source https://www.nuget.org/api/v2/

multiple sources allowed for transient dependencies.

**Comment by forki** [comment]
would this command unzip?

**Comment by pms1969** [comment]
yes that was my thinking, but that could also be a parameter

**Comment by forki** [comment]
but why would you need this then outside the paket.dependencies?

**Comment by pms1969** [comment]
well, one of my specific use cases is in a team wide development tool set.  there's one command to update everything, and I just want to download the packages and unzip them to a specific directory.  Adding another file to the mix just seems silly.  I also quite regularly just download packages to inspect them, and would like to do this with one tool.  just consolidating my toolset as much as possible into as few components as possible.

**Comment by Vilmir** [comment]
We have kind of a similar need. Before doing any package restoration, our codebases are downloading a single bootstrap package and install it in a custom location. Then a .bat is called in the unzipped package to restore the dependencies with Paket.
This bootstrap package is not part of the build dependencies, because it's a prerequisite to fetch those dependencies.

Today we use nuget.exe install to fetch this package. If Paket would have a similar command, we could replace this lonely nuget.exe call.

**Comment by forki** [comment]
First: I'm not against this new command. 

That said I don't really under your use cases. @pms1969 if you have a tool "suite" where the packages have dependencies then I would argue you need the dependencies file and the lock file since paket needs to unify all requirements and come up with a sound resolution. Also having a lock file is really powerful for tool sets, since you see all installed versions in one place. 

@Vilmir how can the one package be a prerequisite? What is it doing?

**Comment by Vilmir** [comment]
Our Sonova.Bootstrapper is a simple GUI tool that wrapped Nuget.Core in the past, and is now wrapping paket:
* Call paket.bootstrapper.exe
* Call paket.exe restore in our Build folder
* Call paket.exe restore in our Source folder

As we have several codebases using the same conventions, we needed to regroup this simple logic in a common tool.

**Comment by forki** [comment]
That makes things pretty complicated if you ask me. Why not using a simple
batch file. Also why are you using 2 restores? The recommended way is to
have 1 dependencies file and maybe multiple groups in it.

**Comment by Vilmir** [comment]
The team has built a nice GUI wrapper for package restore, our developers like it, even if it is just displaying in a pretty console the output of the tools it calls.

We went for 2 sets of dependencies because our architects requested to have a dependency file for the Build system and one for the Source system. The main goal was readability. Our build system is pretty huge and uses a good amount of packages.

**Comment by pms1969** [comment]
for me, it's a commandline tool that autoupdates and then downloads the latest of each tool in our toolchain from our internal nuget servers.  much like the paket bootstrapper or the ```fake boot config``` command.  As I want to keep the tool to just the one file, calling paket fetch from code is preferable to having paket.dependencies and paket.lock files.

all in all, it's not too dissimilar to @Vilmir 's usecase

I'm happy to have a stab at implementing it at some point, just didn't want to do it and have it rejected.  collaboration being good.

**Comment by forki** [comment]
Ok let me formulate it like this: I  think a fetch command can be useful in
some situations. It should also be really straight forward to implement. I
would certainly merge such a PR into v3.

That said: I really think you are making things way to complicated for
yourselves and you should consider to move from nuget mindset over to
embracing dependencies file and lock file (and groups)

**Comment by isaacabraham** [comment]
Can I understand - is this tool / feature either: -

1. A means to an end just to download, one-off, a dependency for e.g. exploration etc.
2. A way of managing dependencies without a lock file etc.?

**Comment by pms1969** [comment]
in my case both, for 2, I'd also consider it a way to  programaticly download some packages that I don't want other artifacts hanging around for.

**Comment by isaacabraham** [comment]
Which artifacts do you mean?

**Comment by Vilmir** [comment]
@forki I agree that we shall adopt a different mindset, migrating out of the Nuget packages.config mess. 
But There is a simple benefit in splitting the build system from the source code dependencies: you can then take the paket.lock file without any treatment and determine the list of artifacts used in your product, a business requirement in our case.

Do you see any upcoming threat in keeping two paket.dependencies files?

**Comment by pms1969** [comment]
@isaacabraham, the paket.lock and paket.dependencies.   In this case, (2) a way to pass in the contents of a dependencies file to the paket install method would be ok (from a programatic perspective).  But not so for use case 1.

**Comment by forki** [comment]
@Vilmir did you look at the docs for groups? If you have one main group and one build group you fulfill these requirements perfectly. see http://fsprojects.github.io/Paket/groups.html

**Comment by Vilmir** [comment]
I did not realize that with groups, the paket.lock file was also segmented nicely.
I'll see if I can change the design to use groups, thanks

**Comment by forki** [comment]
You can also restore independently (if needed) and much more.

**Comment by enricosada** [comment]
`paket add MyPkg -g binaries` can do it.

Will generate paket.dependencies` and `paket.lock`, but behaviour is pretty much what you expect.

Useful for example to download build only packages (like tools)

@pms1969 @Vilmir  can be ok?
